ironfish/src/mining/poolDatabase/database.ts

```diff
@@ -217,8 +230,74 @@ export class PoolDatabase {
       transactionId,
     )
   }
+
+  // Returns a capped number of unique public addresses and the amount of shares
+  // they earned for a specific payout period
+  async payoutAddresses(
+    payoutPeriodId: number,
+  ): Promise<{ publicAddress: string; shareCount: number }[]> {
+    // TODO(mat): This query is very similar to `markSharesPaid`, and if one changes,
+    // the other must change. Consider a way to de-dupe this logic.
+    const sql = `
+      SELECT publicAddress, COUNT(id) shareCount
+      FROM payoutShare
+      WHERE
+        payoutPeriodId = ?
+        AND payoutTransactionId IS NULL
+      GROUP BY publicAddress
+      LIMIT ?
+    `
+    return await this.db.all<{ publicAddress: string; shareCount: number }[]>(
+      sql,
+      payoutPeriodId,
+      MAX_ADDRESSES_PER_PAYOUT,
+    )
+  }
+
+  async markSharesPaid(payoutPeriodId: number, payoutTransactionId: number): Promise<void> {
+    const sql = `
+      UPDATE payoutShare
+      SET payoutTransactionId = ?
+      WHERE
+        payoutPeriodId = ?
+        AND publicAddress IN (
+          SELECT publicAddress
+          FROM payoutShare
+          WHERE
+            payoutPeriodId = ?
+            AND payoutTransactionId IS NULL
+          GROUP BY publicAddress
+          LIMIT ?
+        )
```

Comment:
Are we gonna do this query then fetch the paid shares after this query runs later?

Reply:
Ok, we're gonna try to convert this to a WHERE IN query